importer/likutim/create_unit.go

```diff
@@ -0,0 +1,186 @@
+package likutim
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/Bnei-Baruch/mdb/common"
+	"github.com/Bnei-Baruch/mdb/models"
+	"github.com/Bnei-Baruch/mdb/utils"
+	log "github.com/Sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/volatiletech/sqlboiler/boil"
+	"github.com/volatiletech/sqlboiler/queries/qm"
+	"gopkg.in/volatiletech/null.v6"
+	"io/ioutil"
+	"path"
+	"strings"
+)
+
+type CreateUnits struct {
+	duplicates []Double
+	mdb        *sql.DB
+}
+
+func (c *CreateUnits) Run() {
+	err := c.duplicatesFromJSON()
+	if err != nil {
+		log.Errorf("Error on read. Error: %s", err)
+		compare := new(Compare)
+		compare.Run()
+		c.duplicates = compare.result
+	}
+
+	c.openDB()
+	defer c.mdb.Close()
+
+	c.fetchUnits()
+}
+
+func (c *CreateUnits) fetchUnits() {
+	for _, d := range c.duplicates {
+		log.Debugf("\n\nStart create new unit type LIKUTIM, file: %s all files: %v", d.Save, d.Doubles)
+		fbase, err := models.Files(c.mdb,
+			qm.Where("uid = ?", d.Save),
+		).One()
+		if err != nil {
+			log.Errorf("cant find base file by uid: %s. Error: %s", d.Save, err)
+			continue
+		}
+		cukn, err := models.ContentUnits(c.mdb,
+			qm.Where("id = ?", fbase.ContentUnitID.Int64),
+			qm.Load("Files"),
+		).One()
+		if err != nil {
+			log.Errorf("cant find base unit by file: %v. Error: %s", fbase, err)
+			continue
+		}
+
+		for _, uid := range d.Doubles {
+			tx, err := c.mdb.Begin()
+			if err != nil {
+				log.Errorf("problem open transaction. Error: %s", err)
+				tx.Rollback()
+				continue
+			}
+			f, err := models.Files(c.mdb, qm.Where("uid = ?", uid)).One()
+			if err != nil {
+				log.Errorf("cant find file by uid: %s. Error: %s", uid, err)
+				tx.Rollback()
+				continue
+			}
+
+			u, err := models.ContentUnits(c.mdb, qm.Where("id = ?", f.ContentUnitID.Int64)).One()
+			if err != nil {
+				log.Errorf("cant find unit by file: %v. Error: %s", f, err)
+				tx.Rollback()
+				continue
+			}
+
+			cuo, err := FindOrigin(c.mdb, u.ID)
+			if err != nil {
+				log.Errorf("cant find origin unit by unit: %v. Error: %s", u, err)
+				tx.Rollback()
+				continue
+			}
+
+			cu, err := c.createCU(cuo)
```

Comment:
why are we moving, creating, replacing ?
Do we really need all this ? 
Can't we simply change the type of units from ktaim to likutim ?

Reply:
We still have the mp3 kitei major that will stay in kitei makor units, we move to likutim only the text files

Reply:
I take only doc files from ktaim nivharim, all other must stay on it